core/controller/src/main/scala/whisk/core/controller/Controller.scala

```diff
@@ -180,14 +180,24 @@ object Controller {
         // second argument.  (TODO .. seems fragile)
         val instance = if (args.length > 0) args(1).toInt else 0
 
-        // initialize the runtimes manifest
-        if (config.isValid && ExecManifest.initialize(config)) {
-            val port = config.servicePort.toInt
-            BasicHttpService.startService(actorSystem, "controller", "0.0.0.0", port, new ServiceBuilder(config, instance, logger))
-        } else {
+        def abort() = {
             logger.error(this, "Bad configuration, cannot start.")
             actorSystem.terminate()
             Await.result(actorSystem.whenTerminated, 30.seconds)
         }
+
+        if (!config.isValid) {
+            abort()
+            return
+        }
+
```

Comment:
Exit works too. I accepted the return since this is "main" and in the invoker it is easier to read; and there is more than one "return" in that flow. Good suggestion with sys.exit as an alternative.